packages/read/src/github/github.ts

```diff
@@ -0,0 +1,135 @@
+import axios from 'axios';
+import { GraphData } from '@mindgraph/types';
+import { Provider } from '../types';
+import { Octokit } from 'octokit';
+import { HIDDEN_FILES_REGEX, extractLinksFromLine } from '../common';
+
+const DEFAULT_BRANCH_NAME = 'main';
+const DEFAULT_BASE_PATH = '';
+
+export interface GitHubProviderArgs {
+  token: string;
+  owner: string;
+  repo: string;
+  branch?: string;
+  basePath?: string;
+}
+
+export const GitHub: Provider<GitHubProviderArgs> = {
+  async read({
+    token,
+    owner,
+    repo,
+    branch = DEFAULT_BRANCH_NAME,
+    basePath = DEFAULT_BASE_PATH,
+  }) {
+    const octo = new Octokit({ auth: token });
+
+    const {
+      data: { ref },
+    } = await octo.rest.git.getRef({
+      owner,
+      repo,
+      ref: `heads/${branch}`,
+    });
+
+    return build_graph(octo, { owner, repo, path: basePath, ref });
+  },
+};
+
+type GitHubResult = {
```

Comment:
Question: I'm curious about the internal definition of a `GithubResult` versus something leveraging Octokit's openapi types? https://github.com/octokit/types.ts/issues/267#issuecomment-781637455

Not looking for anything to be changed, just curious.

Reply:
Ya idk lol it's already typed from octokit itself but the type was annoying and I couldn't get it to work so I just wrote a quick one that had what I cared about. Their openapi types would also do the trick

<img width="615" alt="Screenshot 2023-07-03 at 6 28 36 PM" src="https://github.com/jollyjerr/mindgraph/assets/51095001/34db997e-ed90-4710-85be-2c3acffa75ed">